internal/smartlink/smartlink.go

```diff
@@ -131,15 +138,20 @@ func UnlinkContents(src, dest string) error {
 	return nil
 }
 
+func isSymlink(src string) bool {
+	target, err := fileutils.SymlinkTarget(src)
+	return err == nil && src != target
+}
+
 // resolvePaths will resolve src and dest to absolute paths and return them.
 // This is to ensure that we're always comparing apples to apples when doing string comparisons on paths.
 func resolvePaths(src, dest string) (string, string, error) {
 	var err error
-	src, err = fileutils.ResolveUniquePath(src)
+	src, err = filepath.Abs(filepath.Clean(src))
 	if err != nil {
 		return "", "", errs.Wrap(err, "Could not resolve src path")
 	}
-	dest, err = fileutils.ResolveUniquePath(dest)
+	dest, err = filepath.Abs(filepath.Clean(dest))
```

Comment:
Please revert. Absolute path does not give a canonical result, whereas ResolveUniquePath does (or at least -tries to-).